include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -1068,16 +1073,21 @@ struct __early_exit_find_or
             {
                 if constexpr (_OrTagType::value)
                 {
+                    // As far as we find any value of the data here, we can set the atomic value to 1
+                    // No additional actions required.
                     __found_local.store(1);
-                    break;
                 }
                 else
                 {
-                    for (auto __old = __found_local.load(); __comp(__shifted_idx, __old); __old = __found_local.load())
-                    {
-                        __found_local.compare_exchange_strong(__old, __shifted_idx);
-                    }
+                    // As far as we find the first/last value of the data here, we need to set the atomic value to the index of the found data.
+                    // But only in this case when this value is less (if we find the first value)/greater(if we find the last value) than the current value of the atomic.
+                    if constexpr (!_BackwardTagType::value)
+                        __found_local.fetch_min(__shifted_idx);
+                    else
+                        __found_local.fetch_max(__shifted_idx);
                 }
+
+                break;
```

Comment:
Does it make sense to add a comment here about why the break is generally safe? I think especially the backward tag can be confusing if one is unaware that we process elements backwards in this case.

Reply:
Fixed: added additional comments.